### DISCUSSION TITLE
Standardize optional params across repository.

### DISCUSSION THREAD

**Discussion by CCleanerShot** [rad-ui/ui#664] [Q&A] [answered]
As it stands, the project is utilizing 2 different ways to create an optional property:

1)
```ts
interface Props { name: string}

const Test = ({name: ''}: Props) {}
```

2)
```ts
interface Props { name?: string}

const Test = ({ name }: Props) {}
```

Subjective opinion aside, one should be swiftly decided on.

**Comment by kotAPI** [comment]
We should weigh in the pros and cons, I cannot comment on the difference as I do not know whats happening here, lets discuss this on Discord

**Comment by kotAPI** [comment]
@decipher-cs your thoughts?

**Comment by decipher-cs** [comment]
> @decipher-cs your thoughts?

Explicit over implicit. Let's go with number 2 i.e. `interface Props { name?: string}`
This will mean we'll have to put a few extra checks for `null | undefined` here and there but I don't think it should matter much.

**Comment by kotAPI** [comment, marked as answer]
Alright, we'll just define the type and we'll apply the default values on the prop definition of the component then

Something like this, where `hello` is the default value of prop1 for instance
```tsx
type ComponentProps {
  prop1: 'string'
}

const Comp = ({prop1='hello'}:ComponentProps){

}
```
